optuna_dashboard/_app.py

```diff
@@ -411,6 +412,45 @@ def save_study_note(study_id: int) -> BottleViewReturn:
         response.status = 204  # No content
         return {}
 
+    @app.post("/api/studies/<study_id:int>/<trial_id:int>/tell")
+    @json_api_view
+    def tell_trial(study_id: int, trial_id: int) -> BottleViewReturn:
+        s = request.json.get("state", None)
+        vs = request.json.get("values", None)
+
+        try:
+            values = [float(v) for v in vs] if vs is not None else vs
+        except ValueError:
+            response.status = 400  # Bad request
+            return {"reason": "You need to pass float castable values"}
+
+        str_to_state = {
+            "Complete": TrialState.COMPLETE,
+            "Pruned": TrialState.PRUNED,
+            "Fail": TrialState.FAIL,
+        }
+        if s not in str_to_state:
+            response.status = 400  # Bad request
+            return {"reason": f"You passed {s} as a state, which is not defined in Optuna."}
+        state = str_to_state[s]
+
+        try:
+            study_name = storage.get_study_name_from_id(study_id)
+        except KeyError:
+            response.status = 404  # Not found
+            return {"reason": f"study_id={study_id} is not found"}
+
+        study = optuna.load_study(storage=storage, study_name=study_name)
```

Comment:
You don't need to create `Study` class. Could you use `storage` directly.

Reply:
Fixed.